Source/Events.Store.MongoDB/Processing/Streams/StreamProcessorStateDiscriminatorConvention.cs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using MongoDB.Bson;
+using MongoDB.Bson.IO;
+using MongoDB.Bson.Serialization.Conventions;
+
+namespace Dolittle.Runtime.Events.Store.MongoDB.Processing.Streams
+{
+    /// <summary>
+    /// Represents our own custom <see cref="IDiscriminatorConvention"/> used to deal with our MongoDB representation of
+    /// <see cref="StreamProcessorState"/> and <see cref="Partitioned.StreamProcessorStateExtensions"/>.
+    /// </summary>
+    /// <remarks>
+    /// This class was based mostly on information I gathered from these 2 sources:
+    /// https://groups.google.com/forum/#!topic/mongodb-user/iOeEXbUYbo4
+    /// https://github.com/mongodb/mongo-csharp-driver/blob/6b73e381827f83af368a949a807076dad01607fc/MongoDB.DriverUnitTests/Samples/MagicDiscriminatorTests.cs#L53 .
+    /// </remarks>
+    class StreamProcessorStateDiscriminatorConvention : IDiscriminatorConvention
+    {
+        public string ElementName => "Partitioned";
```

Comment:
Make a const string with the name somewhere - so we don't have to rely on typing the same string literal twice.